conans/client/graph/proxy.py

```diff
@@ -59,67 +63,99 @@ def _get_recipe(self, reference, check_updates, update, remotes):
+                                                     new_timestamp=datetime.timestamp(remote_time),
+                                                     new_remote=selected_remote.name)
+                        status = RECIPE_INCACHE_DATE_UPDATED
+                return conanfile_path, status, selected_remote, ref
             else:
-                status = RECIPE_NEWER
+                status = RECIPE_NOT_IN_REMOTE
+                return conanfile_path, status, selected_remote, ref
         else:
             status = RECIPE_INCACHE
+            return conanfile_path, status, cur_remote, ref
+
+    def _get_rrev_from_remotes(self, reference, remotes, check_all_servers):
+        results = []
+        output = ScopedOutput(str(reference), self._out)
+
+        # TODO: cache2.0 --update strategies: when we have specified the revision we don't want to
+        #  check all the remotes, just return the first match
+        output.info(f"Checking all remotes: ({', '.join([remote.name for remote in remotes])})")
+
+        for remote in remotes:
+            try:
+                output.info(f"Checking remote: {remote.name}")
+                remote_rrevs = self._remote_manager.get_recipe_revisions(reference, remote)
+                for rrev in remote_rrevs:
+                    results.append({'remote': remote,
+                                    'reference': reference.copy_with_rev(rrev.get("revision")),
+                                    'time': from_iso8601_to_datetime(rrev.get("time"))})
+                if len(results) > 0 and not check_all_servers:
+                    break
+            except NotFoundException:
+                pass
+
+        if len(results) == 0:
+            return None, None, None
 
-        return conanfile_path, status, selected_remote, ref
+        remotes_results = sorted(results, key=lambda k: k['time'], reverse=True)
+        # get the latest revision from all remotes
+        found_rrev = remotes_results[0]
+        return found_rrev.get("remote"), found_rrev.get("reference"), found_rrev.get("time")
 
-    def _download_recipe(self, ref, output, remotes, remote):
+    # searches in all the remotes and downloads the latest from all of them
+    # TODO: refactor this, it's confusing
+    #  get the recipe selection with _get_rrev_from_remotes out from here if possible
+    def _download_recipe(self, ref, output, remotes, remote, check_all_servers=True,
+                         make_latest=False):
 
-        def _retrieve_from_remote(the_remote):
+        def _retrieve_from_remote(the_remote, reference):
             output.info("Trying with '%s'..." % the_remote.name)
             # If incomplete, resolve the latest in server
-            _ref = self._remote_manager.get_recipe(ref, the_remote)
+            _ref, _ref_time = self._remote_manager.get_recipe(reference, the_remote)
+            new_timestamp = datetime.timestamp(_ref_time) if not make_latest else time.time()
```

Comment:
This time conversion from the remote seems confusing, wouldn't it be possible to communicate with servers with timestamp and not datetimes?